### ISSUE TITLE
Feature request -- enable Backups of Mariadb/MySQL and others

### ISSUE THREAD

**Issue by bd8392** [duplicati/duplicati#4032]
It would be the perfect all in one backup-solution if you were also able to backup and restore databases directly (without scripts directly from the interface). Is there a chance of implementing this? Would be the only known graphical solution for me :-) (besides phpmyadmin which has limitations)

**Comment by ShiromMakkad** [comment]
Can't you just backup all the database's files and restore from that? Why do you need database specific support? Maybe I'm just missing something with how databases work.

**Comment by drwtsn32x** [comment]
@ShiromMakkad - generally with DBMS files you want an application-consistent backup, which requires talking to the database engine.  You can get a backup of the raw files using Duplicati but at best they'd be in a crash-consistent state.

**Comment by bd8392** [comment]
Yes it would be best if duplicatini can talk to the database for the backups...the raw data is not a solution, as they are not consistent... @ShiromMakkad you can backup the database-files, bit that's not the safe or recommended way, because if an application writes to the database, to which duplicati is not talking, the whole backup is broken...

**Comment by ts678** [comment]
[Scripting options](https://duplicati.readthedocs.io/en/latest/06-advanced-options/#scripting-options) can give you a place to do whatever the DBs need to do to prepare for a backup.
This will integrate into the GUI interface (whether done as immediate run, or using a run schedule).
[run-script-example.sh](https://github.com/duplicati/duplicati/blob/master/Duplicati/Library/Modules/Builtin/run-script-example.sh) shows some of what can be done in a script, but simple script is very simple.

**Comment by bd8392** [comment]
ok thanks =) I'll try this =)

**Comment by bd8392** [comment]
Ok I rn into an issue...I backup the database with msqlidump usally...this is only possible inside de mariadb container....so the script can't run inside dolibarr...a remote-backup solution of databases would be nice =)

**Comment by bd8392** [comment]
I'm going to ask this on the forum, as it seems a support questinon =) thanks anyway =)

**Comment by bd8392** [comment]
This doesn't work, as the duplicati-instance is not in the same environment as the MySQL instance (docker). Mysqldump-Script doesn't work, as mysql has to be installed...I'm not aware of other databases, but this feature would be extremely useful :-)

**Comment by bd8392** [comment]
https://forum.duplicati.com/t/mariadb-mysql-backups-with-duplicati/9613

**Comment by ts678** [comment]
So this DB is in Docker. Is Duplicati also in Docker (different one), or on the host system?

Either way, Duplicati can run a script which would get the DB to mysqldump somewhere.
After the DB Docker has finished preparing a dump of the DB, then Duplicati backs it up.

When you design this system, keep in mind what tools you may need if you ever restore.
You will also need to develop a procedure based on this system design (Docker and DB).

As a side note, Duplicati will probably not deduplicate these dumps much, as they sound
like the data will tend to move around from dump to dump, defeating re-use of blocks...

The Mariabackup or Percona XtraBackup tools used well might possibly save space, if the
plan is for Duplicati to do remote backup of whatever the local tool puts on local storage.
Incrementals will probably be small, whereas a new mysqldump will probably be full-size.

I don't use Docker, and cannot help beyond what the Docker experts say about remotes.
Web search shows quite a few discussions of how to get in a container to get things run.
Run Google search for `run command in another Docker` (for example) for some thoughts.

Is this MariaDB or MySQL or TBD? I don't use any of them, so will link to documentation:

[Backing Up and Restoring Databases](https://mariadb.com/kb/en/backing-up-and-restoring-databases/) is MariaDB info.
[Chapter 7 Backup and Recovery](https://dev.mysql.com/doc/refman/8.0/en/backup-and-recovery.html) is MySQL info.

**Comment by bd8392** [comment]
> So this DB is in Docker. Is Duplicati also in Docker (different one), or on the host system?

Yes the DB is on the same system, but in different container...so a "remote backup" out of the duplicati-container would be required, which is not possible, because mysqld is not installed in the container...so all scripts with the mysqldump -h will fail in duplicati...

The run command in docker also doesn't work, as I run my containers in swarm... this leads to unpredictable container names on the node. So if I right a command, it fails on next reboot, because the name of the container is different...

My goal was, to use duplicati as a all-in-one backup solution, which is not possible without this feature in my situation :-). It's still very "unhandy" to backup databases in general, because in most cases, you need scripts, which are only working on the database-container... therefore a "source option" for different databases would be perfect! :-).

Thanks for mentioning the other programs :-) I'll look into them :-). For now I'll use them to backup the database to a local folder, and duplicati pushes them to remote-folder...
